Index: lib/erector/rails/extensions/widget.rb
===================================================================
--- lib/erector/rails/extensions/widget.rb	(revision 185)
+++ lib/erector/rails/extensions/widget.rb	(working copy)
@@ -1,10 +1,15 @@
 module Erector
   Widget.class_eval do
     include ActionController::UrlWriter
-    attr_reader :_erbout
-
-    after_initialize do
-      @_erbout = doc.string
+    
+    def _erbout
+      unless @_erbout 
+        def doc.<< s
+          output << s
+        end
+        @_erbout = doc 
+      end
+      @_erbout
     end
 
     def fake_erbout
@@ -56,6 +61,7 @@
     # return text, take block
     [
       :link_to_function,
+      :hidden_field_tag,
       :text_field_tag,
       :password_field_tag,
       :check_box_tag
@@ -118,4 +124,4 @@
       helpers.pluralize(*args)
     end    
   end
-end
\ No newline at end of file
+end
